src/metabase/models/pulse_card.clj

```diff
@@ -1,4 +1,42 @@
 (ns metabase.models.pulse-card
-  (:require [toucan.models :as models]))
+  (:require [metabase.util :as u]
+            [metabase.util.schema :as su]
+            [schema.core :as s]
+            [toucan
+             [db :as db]
+             [models :as models]]))
 
 (models/defmodel PulseCard :pulse_card)
+
+(defn next-position-for
+  "Return the next available `pulse_card.position` for the given `pulse`"
+  [pulse-id]
+  {:pre [(integer? pulse-id)]}
+  (-> (db/query {:select [:%max.position]
```

Comment:
Why not either `db/select-one-field` or also do the increment in SQL?

Reply:
Maybe I'm doing it wrong, but I can't make `select-one-field` work with the function:

```
(db/select-one-field :%max.position PulseCard :pulse_id pulse-id)
```

does not work

Reply:
That doesn't work -- it's a limitation of Toucan. Do this instead

```clj
(:max (db/select-one [PulseCard [:%max.position :max]] :pulse_id pulse-id))
```